src/main/java/org/csanchez/jenkins/plugins/kubernetes/ConfigUtils.java

```diff
@@ -0,0 +1,37 @@
+package org.csanchez.jenkins.plugins.kubernetes;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class ConfigUtils {
+	private static final Logger LOGGER = Logger.getLogger(ConfigUtils.class.getName());
+	public static final String KUBERNETES_NAMESPACE_PATH = "/var/run/secrets/kubernetes.io/serviceaccount/namespace";
+
+	public static String getNamespace(String namespace) {
+		LOGGER.log(Level.FINEST,
+				"getNamespace called with "+namespace);
+		if (namespace != null && !namespace.equals("")) {
```

Comment:
yeah the intent was to be able to blindly call this if the calling code didn't know if it already had a namespace or not.  If it already had a valid value, we'd short-circuit looking at the filesystem for a value.

I may revisit that pattern as it's a bit ugly.  But it was intentional.